### PR TITLE
Convert ast-view to typescript

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
Moves the ast view pane to TS

### CODE REVIEW

File: static/panes/ast-view.ts [hunk 1]
```diff
@@ -0,0 +1,284 @@
+// Copyright (c) 2017, Simon Brand
```
Comment (jeremy-rifkin) [inline static/panes/ast-view.ts:1]: Quick question: Should this line be updated or left with the original copyright from ast-view.js?

Comment (mattgodbolt) [reply static/panes/ast-view.ts:1]: I think it should be left alone. We can ask @TartanLlama if they'd like us to update the attribution, but for best I'd leave the copyright alone wtihout clear intent from the original author. Thanks!

File: static/panes/ast-view.ts [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+'use strict';
+
+import _ from 'underscore';
```
Comment (RubenRBS) [inline static/panes/ast-view.ts:27]: Can you try to drop the underscore dependency? TS has support for most of the things that we need from _

File: static/panes/ast-view.ts [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+import $ from 'jquery';
```
Comment (RubenRBS) [inline static/panes/ast-view.ts:28]: This is already globally defined and can be dropped

File: static/panes/ast-view.ts [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+'use strict';
```
Comment (RubenRBS) [inline static/panes/ast-view.ts:25]: This is set by default when compiling ts code so no need to specify it and can be dropped

File: static/panes/ast-view.ts [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+            lineNumbersMinChars: 3,
+        }));
+    }
+    
+    override getDefaultPaneName() {
+        return 'Ast Viewer';
+    }
+
+    onMouseMove(e: monaco.editor.IEditorMouseEvent) {
+        if (e === null || e.target === null || e.target.position === null) return;
```
Comment (RubenRBS) [inline static/panes/ast-view.ts:126]:
```suggestion
        if (e?.target?.position === null) return;
```

**Review by RubenRBS — COMMENTED** [review]
Thank you for taking the time to port this pane :)

**Comment by jeremy-rifkin** [comment]
Alright, I've implemented the feedback. I removed most uses of underscore.js but it is still needed at least for `_.throttle`. (The remaining use of underscore.js is consistent ir-view.ts which I've been using as a reference for the conversion).

**Review by matsjla — APPROVED** [review]
LGTM - There is some duplicated code with the color/pane linking (with ir-view.ts), but I'll figure out a nice way to deal with that - Thanks!

**Pull request merged**